src/pybind/mgr/dashboard/frontend/src/app/ceph/nfs/nfs-form/nfs-form.component.ts

```diff
@@ -127,12 +131,22 @@ export class NfsFormComponent extends CdForm implements OnInit {
 
     if (this.isEdit) {
       this.action = this.actionLabels.EDIT;
-      this.route.params.subscribe((params: { cluster_id: string; export_id: string }) => {
-        this.cluster_id = decodeURIComponent(params.cluster_id);
-        this.export_id = decodeURIComponent(params.export_id);
-        promises.push(this.nfsService.get(this.cluster_id, this.export_id));
-        this.getData(promises);
-      });
+      this.route.params.subscribe(
+        (params: { cluster_id: string; export_id: string; rgw_export_type?: string }) => {
+          this.cluster_id = decodeURIComponent(params.cluster_id);
+          this.export_id = decodeURIComponent(params.export_id);
+          if (params.rgw_export_type) {
+            this.nfsForm.get('rgw_export_type').setValue(params.rgw_export_type);
+            if (params.rgw_export_type === 'bucket') {
+              this.resolveBucket();
```

Comment:
We should follow the same pattern as existing one.
Like adding promises then resolve in getData() as right now its confusing to see which data is from where.
This is a complex form so lets try to keep it consistent in a way easy to follow.

Reply:
Well the purpose here is different then what's done in `getData()`. There its supposed to be loaded on ngOnInit(), but I would prefer to avoid subscribing to rgw user/bucket APIs if its lets say Cephfs export & only fetch the data when we actually need it , which is when a user selects the particular radio button